### PR TITLE
alternative implementation of *format-display-name* hook

### PR DESCRIPTION
**Pull request by darwin**
@roman01la this is just for your consideration, feel free to scratch it

### CODE REVIEW

**Review by roman01la — COMMENTED** [review]
I like this better, thanks.

File: core/src/uix/compiler/alpha.cljs [hunk 1]
```diff
@@ -352,10 +352,14 @@
...
 
+(set! *format-display-name* default-format-display-name)
+
 (defn format-display-name [^string s]
-  (if *format-display-name*
-    (*format-display-name* s default-format-display-name)
-    (default-format-display-name s)))
+  (if (fn? *format-display-name*)
+    (*format-display-name* s)
+    (throw (ex-info "unexpected uix.compiler.alpha/*format-display-name* is not bound to a function"
```
Comment (roman01la) [inline core/src/uix/compiler/alpha.cljs:360]: So we really need it here? It's probably user's responsibility to not assign something other than a function

**Comment by darwin** [comment]
Just a sanity check, I'm ok if you want to remove it. But leave there at least assert so the stack trace is better than infamous ".call to undefined".

File: core/src/uix/compiler/alpha.cljs (hunk 1 above)
Comment (darwin) [reply core/src/uix/compiler/alpha.cljs:360]: Just a sanity check, I'm ok if you want to remove it. But leave there at least assert so the stack trace is better than infamous ".call to undefined".

Comment (roman01la) [reply core/src/uix/compiler/alpha.cljs:360]: I think in this case the var will be always bound because it is initialized with the default function.

Comment (roman01la) [reply core/src/uix/compiler/alpha.cljs:360]: I've changed the var to be initialized with a default fn, but you are right about ".call to undefined", saw this in your tests.

**Pull request merged**